internal/fwtype/missing_underlying_type_validation_test.go

```diff
@@ -0,0 +1,1611 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package fwtype_test
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/internal/fwtype"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func TestContainsMissingUnderlyingType(t *testing.T) {
```

Comment:
nit: Do we want to add a test here with a custom type in a collection/object/etc?

Reply:
Is https://github.com/hashicorp/terraform-plugin-framework/pull/991/commits/584f83db2e23e9ec145d9377391836f9dbad583f what you are looking for?

Reply:
Yeah that looks great!